Adding Cognito examples for SRP Authentication

bhatti opened the pull request:
# aws-doc-sdk-examples Pull Request

Thank you for making a submission to the *aws-doc-sdk-examples* repository. For more information about submitting pull requests to this repository, see [Guidelines for contributing](https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/CONTRIBUTING.md).

*NOTE:* This pull request (PR) template contains two sections. Use the section that applies to your submission, and remove the section which doesn't apply.
***
## I'm submitting a new SDK code example

Confirm you have met the following minimum requirements:

- [ X] Add the following copyright notice to the top of each file. See the [LICENSE agreement](https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/LICENSE) for the terms of the copyright. 
>Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.\
>SPDX-License-Identifier: Apache-2.0
- [ ] Create unit tests for all paths through the code, and confirm that the all the tests pass. For recommendations, see [How we test code examples](https://github.com/awsdocs/aws-doc-sdk-examples/wiki/Code-quality-guidelines---testing-and-linting#how-we-test-code-examples).
- [ X] Run a linter against all code and implement the resulting suggestions. For recommendations, see [Linters run on check in](https://github.com/awsdocs/aws-doc-sdk-examples/wiki/Code-quality-guidelines---testing-and-linting#linters-run-on-check-in).
- [X ] Add minimum usage documentation as comments in the code. For recommendations, see [Code comment guidelines](https://github.com/awsdocs/aws-doc-sdk-examples/wiki/Code-comment-guidelines).
***
## I'm resolving an issue with an existing code example

Describe the changes you have made here, including any issue numbers.
Adding changes that Feng provided for SRP integration with Cognito.

Confirm you have met the following minimum requirements:

- [X ] Test the changed code. For recommendations, see [How we test code examples](https://github.com/awsdocs/aws-doc-sdk-examples/wiki/Code-quality-guidelines---testing-and-linting#how-we-test-code-examples).
- [ X] Run a linter against the changed code and implement the resulting suggestions. For recommendations, see [Linters run on check in](https://github.com/awsdocs/aws-doc-sdk-examples/wiki/Code-quality-guidelines---testing-and-linting#linters-run-on-check-in).
***
## Open source license adherence

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

AMZ-brandon commented:
Hey code examples team! Brandon here, Cognito tech writer. This set of examples was provided by our Cognito team and would be a valuable addition to the library.

scmacdon commented:
Please reach out to our SDK Editor Liz Kellebrew-Davies who must edit this PR before it can be merged.

ford-at-aws reviewed (approved):
lgtm

ford-at-aws commented:
Adding exception for 40-character violation.

lkdavies commented on javav2/example_code/cognito/src/main/java/com/example/cognito/srp/usecases/DeviceAuthDemo.java:
```diff
@@ -0,0 +1,109 @@
+        BigInteger A = clientKeys.right();
+        RespondToAuthChallengeResponse deviceSrpChallengeResponse = respondToDeviceSrpAuthChallenge(username, deviceKey, A.toString(16));
+        System.out.println("DeviceSrpAuth response: " + deviceSrpChallengeResponse);
+        // Get response from SRP initial auth
```
Add ending punctuation.

lkdavies commented on javav2/example_code/cognito/src/main/java/com/example/cognito/srp/usecases/UserAuthDemo.java:
```diff
@@ -0,0 +1,111 @@
+        BigInteger A = clientKeys.right();
+        InitiateAuthResponse initiateAuthResponse = initiateAuth(username, A.toString(16));
+        System.out.println("InitiateAuth response: " + initiateAuthResponse);
+        // Get response from SRP initial auth
```
Add ending punctuation.

bhatti replied:
Updated comments with ending punctuation.

lkdavies reviewed (commented):
Editorial comments

bhatti commented on javav2/example_code/cognito/src/main/java/com/example/cognito/srp/SRPDemo.java:
```diff
@@ -0,0 +1,136 @@
+            .build();
+
+    static {
+        // Create user pool
```
Updated comments with ending punctuation.

bhatti commented on javav2/example_code/cognito/src/main/java/com/example/cognito/srp/SRPDemo.java:
```diff
@@ -0,0 +1,136 @@
+                .softwareTokenMfaConfiguration(SoftwareTokenMfaConfigType.builder().enabled(true).build())
+                .build();
+        cognitoClient.setUserPoolMfaConfig(mfaRequest);
+        // Create user application client
```
Updated comments with ending punctuation.

bhatti commented on javav2/example_code/cognito/src/main/java/com/example/cognito/srp/SRPDemo.java:
```diff
@@ -0,0 +1,136 @@
+        CreateUserPoolClientResponse userClientResponse = cognitoClient.createUserPoolClient(userClientRequest);
+        COGNITO_USER_CLIENT_ID = userClientResponse.userPoolClient().clientId();
+        System.out.println("Application client for user is: " + COGNITO_USER_CLIENT_ID);
+        // Create admin application client
```
Updated comments with ending punctuation.

bhatti commented on javav2/example_code/cognito/src/main/java/com/example/cognito/srp/SRPDemo.java:
```diff
@@ -0,0 +1,136 @@
+        COGNITO_ADMIN_CLIENT_ID = adminClientResponse.userPoolClient().clientId();
+        COGNITO_ADMIN_CLIENT_SECRET = adminClientResponse.userPoolClient().clientSecret();
+        System.out.println("Application client for admin is: " + COGNITO_ADMIN_CLIENT_ID);
+        // Create user
```
Updated comments with ending punctuation.

bhatti commented on javav2/example_code/cognito/src/main/java/com/example/cognito/srp/SRPDemo.java:
```diff
@@ -0,0 +1,136 @@
+    }
+
+    public static void main(String[] args) {
+        // 1. Start authentication with Admin credentials
```
Updated comments with ending punctuation.

bhatti commented on javav2/example_code/cognito/src/main/java/com/example/cognito/srp/usecases/DeviceAuthDemo.java:
```diff
@@ -0,0 +1,109 @@
+        BigInteger salt = new BigInteger(deviceSrpChallengeResponse.challengeParameters().get("SALT"), 16);
+        String secretBlock = deviceSrpChallengeResponse.challengeParameters().get("SECRET_BLOCK");
+
+        // Calculate password claim signature
```
Updated comments with ending punctuation.

bhatti commented on javav2/example_code/cognito/src/main/java/com/example/cognito/srp/usecases/UserAuthDemo.java:
```diff
@@ -0,0 +1,111 @@
+        String userId = initiateAuthResponse.challengeParameters().get("USER_ID_FOR_SRP");
+        String secretBlock = initiateAuthResponse.challengeParameters().get("SECRET_BLOCK");
+
+        // Calculate password claim signature
```
Updated comments with ending punctuation.

bhatti commented:
@lkdavies, I added punctuation to the comments and changed the size of the identifier exceeding 40 characters. Please take a look at the updated commits.
